Fast-DDS uses SHM transport by default, so network traffic cannot be … (#3683)
ros2/ros2_documentation: source/Tutorials/Advanced/Security/Examine-Traffic.rst

Prior tutorials enabled security, but how can you **really** tell if traffic is being encrypted?
In this tutorial we'll take a look at capturing live network traffic to show the difference between encrypted and unencrypted traffic.

.. note::

>   Please go through this tutorial to use different host system for talker and listener with ``rmw_fastrtps_cpp``.

Author: i think that is okay to ask this tutorial with different machines, since it has been already explained previous tutorial page, https://docs.ros.org/en/rolling/Tutorials/Advanced/Security/Security-on-Two.html

>   Please go through this tutorial to use different host system for talker and listener with ``rmw_fastrtps_cpp``.

Reviewer: I don't we should force those who use `rmf_fastrtps_cpp` to go over this tutorial on multiple systems. We can update the note to inform users that if they are testing with `rmf_fastrtps_cpp`, they should be mindful that SHM is enabled by default which would not allow them to capture UDP packets. They can either 1) Switch to two hosts or 2) Disable SHM by creating an xml config that uses UDP for transport as seen [here](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/udp/udp.html#enabling-udp-transport) and further letting ROS 2 know to use this xml config as described [here](https://github.com/ros2/rmw_fastrtps#full-qos-configuration)

Author: sounds good to me. addressed in https://github.com/ros2/ros2_documentation/pull/3683/commits/2eb2e66ec9242105424a48d33707c4c207435871

  ``rmw_fastrtps_cpp`` uses `Shared Memory Transport <https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/shared_memory/shared_memory.html>`_ by default to improve the performance in the transport layer when the endpoints are in the same host system.

>   Security enclaves still can be applied, effective and data will be encrypted, but we cannot capture live network traffic since the data will not be on the network interface.

Author: note that, this is just transport improvement, not data sharing like zero copy. data will be serialized and encrypted.

Prior tutorials enabled security, but how can you **really** tell if traffic is being encrypted?
In this tutorial we'll take a look at capturing live network traffic to show the difference between encrypted and unencrypted traffic.

.. note::

>   ``rmw_fastrtps_cpp`` uses `Shared Memory Transport <https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/shared_memory/shared_memory.html>`_ by default to improve the performance in the transport layer when the endpoints are in the same host system.
>   Because of this improvement, we cannot capture live network traffic since the data will not be on the network interface.
>   In case of  ``rmw_fastrtps_cpp``, we need to do either going through this tutorial to use different host system or disabling shared memory transport with `Enabling UDP Transport <https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/udp/udp.html#enabling-udp-transport>`_ and `How to set Fast-DDS XML configuration <https://github.com/ros2/rmw_fastrtps#full-qos-configuration>`_.
>   Security enclaves still can be applied, effective and data will be encrypted.

Reviewer 2: ```suggestion
  ``rmw_fastrtps_cpp`` uses `Shared Memory Transport <https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/shared_memory/shared_memory.html>`_ by default to improve the performance in the transport layer when the endpoints are in the same host system.
  Security enclaves are still applied, and data will be encrypted.
  However, you cannot capture live network traffic since the data will not be on the network interface.
  If you are using  ``rmw_fastrtps_cpp``, you need to either go through this tutorial and use a different host system between the publisher and subscriber, or disable shared memory transport with `Enabling UDP Transport <https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/udp/udp.html#enabling-udp-transport>`_ and `How to set Fast-DDS XML configuration <https://github.com/ros2/rmw_fastrtps#full-qos-configuration>`_.
```

Author: addressed in https://github.com/ros2/ros2_documentation/pull/3683/commits/0e272d3a59749df51121a69488bddf776d464e74

.. code-block:: bash

  # Disable ROS Security for both terminals
  export ROS_SECURITY_ENABLE=false

> 

Reviewer 3: It's good to test by setting ROS_SECURITY_ENABLE with `false` explicitly, but the above doc says `not set`.

Author: addressed in https://github.com/ros2/ros2_documentation/pull/3683/commits/35d516001d109235c129286bb295a9a397e76f35
